apps/smart-sheet/src/inngest/functions/users/sync-users-page.ts

```diff
@@ -0,0 +1,110 @@
+import type { User } from '@elba-security/sdk';
+import { Elba } from '@elba-security/sdk';
+import { eq } from 'drizzle-orm';
+import { NonRetriableError } from 'inngest';
+import { type SmartSheetUser, getUsers } from '@/connectors/users';
+import { db } from '@/database/client';
+import { Organisation } from '@/database/schema';
+import { env } from '@/env';
+import { inngest } from '@/inngest/client';
+
+const elbaUsers = (userData: SmartSheetUser[]) => {
+  return userData
+    .filter((user) => user.status === 'ACTIVE')
+    .map(
+      (user: SmartSheetUser): User => ({
+        id: JSON.stringify(user.id),
+        displayName: user.name,
+        email: user.email,
+        additionalEmails: [],
+      })
+    );
+};
+
+/**
+ * DISCLAIMER:
+ * This function, `syncUsersPage`, is provided as an illustrative example and is not a working implementation.
+ * It is intended to demonstrate a conceptual approach for syncing users in a SaaS integration context.
+ * Developers should note that each SaaS integration may require a unique implementation, tailored to its specific requirements and API interactions.
+ * This example should not be used as-is in production environments and should not be taken for granted as a one-size-fits-all solution.
+ * It's essential to adapt and modify this logic to fit the specific needs and constraints of the SaaS platform you are integrating with.
+ */
+export const syncUsersPage = inngest.createFunction(
+  {
+    id: 'sync-users-page',
+    priority: {
+      run: 'event.data.isFirstSync ? 600 : 0',
+    },
+    concurrency: {
+      key: 'event.data.organisationId',
+      limit: 100,
+    },
+    retries: 1,
+  },
+  { event: 'smart-sheet/users.page_sync.requested' },
+  async ({ event, step }) => {
+    /* eslint-disable -- no type here */
+    const { organisationId, syncStartedAt, page, region } = event.data;
+
+    const elba = new Elba({
+      organisationId,
+      apiKey: env.ELBA_API_KEY,
+      baseUrl: env.ELBA_API_BASE_URL,
+      region,
+    });
+
+    // retrieve the SaaS organisation token
+    const token = await step.run('get-token', async () => {
+      const [organisation] = await db
+        .select({ token: Organisation.accessToken })
+        .from(Organisation)
+        .where(eq(Organisation.id, organisationId));
+
+      if (!organisation) {
+        throw new NonRetriableError(`Could not retrieve organisation with id=${organisationId}`);
+      }
+      return organisation.token;
+    });
+
+    const nextPage = await step.run('list-users', async () => {
+      // retrieve this users page
+      const result: any = await getUsers(token, page);
```

Comment:
```suggestion
      const result = await getUsers(token, page);
```

`any` is forbidden in the tsconfig

Reply:
Changes Done